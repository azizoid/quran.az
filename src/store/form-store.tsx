import { createContext, useEffect, useMemo, useState } from 'react'

import { useRouter } from 'next/router'

import { FormProps } from '@/lib/types'
import { getView } from '@/utility'

export const FormContext = createContext({} as FormProps)

type FormContextProviderProps = {
  children?: React.ReactNode
}

export const FormContextProvider = ({ children }: FormContextProviderProps) => {
  const { query } = useRouter()
  const [state, setState] = useState<FormProps>({} as FormProps)

  const getDataFromRouter = useMemo(
    () =>
      getView({
        s: Number(query?.soorah?.toString()) || 0,
        a: Number(query?.ayah?.toString()) || '',
        q: query?.search?.toString() || '',
        t: Number(query?.t?.toString()),
      }),
    [query?.ayah, query?.search, query?.soorah, query?.t]
  )

  useEffect(() => {
    setState(getDataFromRouter)
  }, [getDataFromRouter])

  return <FormContext.Provider value={state}>{children}</FormContext.Provider>
}
