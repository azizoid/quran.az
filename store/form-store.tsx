import { createContext, FC, useEffect, useMemo, useState } from 'react'
import { useRouter } from 'next/router'
import { FormProps } from '@/lib/types'

import { getView } from '../utility/getView/getView'

export const FormContext = createContext({} as FormProps)

type FormContextProviderProps = {
  children?: React.ReactNode
}

export const FormContextProvider: FC<FormContextProviderProps> = ({ children }) => {
  const { query } = useRouter()
  const [state, setState] = useState<FormProps>()

  const getDataFromRouter = useMemo(
    () =>
      getView({
        s: Number(query?.soorah?.toString()) || 0,
        a: Number(query?.ayah?.toString()) || '',
        q: query?.search?.toString() || '',
        t: Number(query?.t?.toString()),
      }),
    [query?.ayah, query?.search, query?.soorah, query?.t]
  )

  useEffect(() => {
    setState(getDataFromRouter)
  }, [getDataFromRouter])

  return <FormContext.Provider value={state}>{children}</FormContext.Provider>
}
