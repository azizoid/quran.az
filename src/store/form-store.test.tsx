import { render } from '@testing-library/react'
import { useRouter } from 'next/router'

import { FormContextProvider } from './form-store'

const push = jest.fn()
jest.mock('next/router', () => ({
  useRouter: jest.fn(),
}))

test('FormContextProvider snapshot', () => {
  ;(useRouter as jest.Mock).mockImplementation(() => ({
    push,
  }))

  const { container } = render(<FormContextProvider />)

  expect(container).toMatchSnapshot()
})

test('FormContextProvider snapshot', () => {
  ;(useRouter as jest.Mock).mockImplementation(() => ({
    query: { soorah: 23, ayah: 23, search: 'Musa', t: 1 },
    push,
  }))

  const { container } = render(<FormContextProvider />)

  expect(container).toMatchSnapshot()
})
