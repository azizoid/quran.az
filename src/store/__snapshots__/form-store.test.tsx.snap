// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormContextProvider snapshot 1`] = `<div />`;

exports[`FormContextProvider snapshot 2`] = `<div />`;
